File: my-folder/0053-maximum-subarray/solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max = Integer.MIN_VALUE;
        int currentMax = 0;
        for (int i = 0; i < nums.length; i++) {
            currentMax += nums[i];
            if (currentMax > max) {
                max = currentMax;
            }
            if (currentMax < 0) {
                currentMax = 0;
            }
        }
        return max;
    }
}

File: my-folder/0053-maximum-subarray/solution.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        current_sum = 0
        largest_sum = -999999
        for num in nums:
            current_sum += num
            if current_sum > largest_sum:
                largest_sum = current_sum
            if current_sum < 0:
                current_sum = 0
        return largest_sum
